apiary.apib

```diff
@@ -1036,6 +1036,332 @@ Outputs the list of peers that were removed.
 
 # Group refs
 
+Retrieves the object named by <ipfs-path> and displays the link
+hashes it contains.
+
+## refs [GET /refs{?arg}{&format,edges,unique,recursive}]
+Lists links (references) from an object
+
+#### curl
+    curl -i http://localhost:5001/api/v0/refs?arg=/ipfs/QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ/cat.jpg
+
++ Parameters
+    + arg (string, required) - Path to the object or objects to list refs from.
+    + format (string, optional) - Emit edges with given format. Tokens: `<src> <dst> <linkname>`.
+    + edges (boolean, optional) - Emit edge format: `<from> -> <to>`. Alias: e.
+    + unique (boolean, optional) - Omit duplicate refs from output. Alias: u.
+    + recursive (boolean, optional) - Recursively list links of child nodes. Alias: r.
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/refs"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/refs"
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Fri, 05 Feb 2016 17:30:09 GMT
+        Content-Length: 32
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        Argument 'ipfs-path' is required
+        ```
+
++ Request With Empty Argument
+
+    The response is the same if there is an invalid argument. For example:
+
+        curl -i "http://localhost:5001/api/v0/refs?arg=kitten"
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/refs?arg="
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/refs?arg="
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 05 Feb 2016 17:30:51 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "invalid ipfs ref path"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "invalid ipfs ref path",
+          "Code": 0
+        }
+        ```
+
++ Request With Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/refs?arg=QmPEKipMh6LsXzvtLxunSPP7ZsBM8y9xQ2SQQwBXy5UY6e"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/refs?arg=QmPEKipMh6LsXzvtLxunSPP7ZsBM8y9xQ2SQQwBXy5UY6e"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Chunked-Output: 1
+        Date: Fri, 05 Feb 2016 17:31:52 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes
+
+    + Body
+
+        ```
+        ```
```

Comment:
The optional description section for the response. I could just say, in instances like this, 'There is no body'